Create questions and answers

kkan opened the pull request:
Added controllers and simple views

kkan committed:
gemfile changes and generators config

kkan committed:
answer presence of question validation

kkan committed:
ability to create and view question

kkan committed:
answers creating functionality

vkurennov commented on app/controllers/answers_controller.rb:
```diff
@@ -0,0 +1,13 @@
+  def create
+    @question = Question.find(params[:question_id])
+    ans = @question.answers.create(answer_params)
+    redirect_to @question
```
Если будут ошибки валидации, то при редиректе они исчезнут и также сотрется все, что пользователь ввел. Поэтому, для вывода ошибок нужен рендер. Другой вопрос, что тут нужно рендерить, если форма ответа находится на странице вопроса.

vkurennov commented on app/models/answer.rb:
```diff
@@ -1,5 +1,5 @@
   belongs_to :question
 
-  validates :body, presence: true
+  validates :body, :question, presence: true
```
Лучше валидировать question_id, чтобы не грузить ассоциацию из базы. А если используешь 5-е рельсы, то в этом нет неободимости, т.к. по-умолчанию, ассоциация belongs_to теперь автоматом валидирует наличие связанного объекта

vkurennov commented on spec/controllers/answers_controller_spec.rb:
```diff
@@ -0,0 +1,30 @@
+    describe 'valid params' do
+      it 'creates answer in db' do
+        expect { post :create, params: { question_id: question, answer: attributes_for(:answer) } }.
+          to change(Answer, :count).by(1)
```
Тут надо проверить, что ответ связывается с вопросом. Можно это сделать сразу в change через ассоциации вопроса

vkurennov reviewed (approved):
Принимаю, но исправь замечания при выполнении следующего задания. И начни как раз с них.

kkan committed:
Adding answer logic changes.
Remove presence validation for question (rails 5 does it).
Gem pry added.
Setting new answer in question page.
Some routes changes.
